sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts

```diff
@@ -64,4 +79,96 @@ export class InstitutionAESTController extends BaseController {
     }
     await this.institutionService.updateInstitution(institutionId, payload);
   }
+
+  /**
+   * Controller method to get all institution users with the
+   * given institutionId.
+   * @queryParm page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @queryParm pageLimit, page size or records per page, if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @queryParm searchName, user's name keyword to be searched
+   * @queryParm sortField, field to be sorted
+   * @queryParm sortOrder, order to be sorted
+   * @returns All the institution users for the given institution
+   * with total count.
+   */
+  @Get(":institutionId/user")
+  async getInstitutionUsers(
+    @Param("institutionId") institutionId: number,
+    @Query(PaginationParams.SearchCriteria) searchName: string,
+    @Query(PaginationParams.SortField) sortField: string,
+    @Query(PaginationParams.SortOrder) sortOrder = FieldSortOrder.ASC,
+    @Query(PaginationParams.Page) page = DEFAULT_PAGE_NUMBER,
+    @Query(PaginationParams.PageLimit) pageLimit = DEFAULT_PAGE_LIMIT,
+  ): Promise<PaginatedResults<InstitutionUserAPIOutDTO>> {
+    return this.institutionControllerService.getInstitutionUsers(
+      institutionId,
+      {
+        page: page,
+        pageLimit: pageLimit,
+        searchCriteria: searchName,
+        sortField: sortField,
+        sortOrder: sortOrder,
+      },
+    );
+  }
+
+  /**
+   * Search the institution based on the search criteria.
+   * @param legalName legalName of the institution.
+   * @param operatingName operatingName of the institution.
+   * @returns Searched institution details.
+   */
+  @Get("all/search")
```

Comment:
Is the "all" added here to avoid route conflicts? Is it part of the refactor or was it already there?
Do we really need it?

Reply:
Yes it was introduced to avoid route conflicts. This was conflicting with @Get(":institutionId") in the same controller.
Let me add that to PR description.

Reply:
There is a way to order the methods to avoid it. If I am not wrong @ann-aot did it in another PR to resolve the same scenario.

Reply:
ya , you can this is controller before the `@Get(":institutionId")` controller

Reply:
maybe you can also add a comment

Reply:
Thanks for letting know @ann-aot. Will do that.